R/executionSettings.R

```diff
@@ -0,0 +1,198 @@
+
+# ExecutionSettings ----
+
+#' @description
+#' An R6 class to define an ExecutionSettings object
+#'
+#' @export
+ExecutionSettings <- R6::R6Class(
+  classname = "ExecutionSettings",
+  public = list(
+    initialize = function(connectionDetails = NULL,
```

Comment:
Just FYI, the from of initialize here is different than what we discussed about R6 classes in the monthly meeting:

1. Initialize takes list-of-list (ie: fromJSON()) or a string which is JSON format and the object is initialized from that.
2. The objective of those R6 classes (in CohortIncidence) was to just enshroud object state into a formalized construct.   So, it wasn't meant to have behavior about connecting, and the entire object model should be able to be serialized from/to json and I'm not sure connection details is that.

So based on the above I was imagining the execution settings to contain user-input data about schemas, runtime options, etc, and then to execute a study you have the analysis, execution settings and connection details provided separately (especially because sometimes you pass along an actual open connection (for whatever reason) or a connection details and the code decides which to use.

Reply:
I think the context of this class is useful for a different reason that serialization or de-serialization. This would allow us to standardize across packages how we encapsulate inputs. At the moment these are fairly inconsistent.

I would say we explicitly never want to serialize connectionDetails as this adds the potential to leak security details. So the construction should be:

```{r}
createExecutionSettings <- function(connection = NULL, connectionDetails = NULL, cdmDatabaseSchema, cohortDatabaseSchema = ... etc) {
   data <- list(
      <insertAllCommonSettings>
   )
   ExecutionSettings$new(connection = connection, connectionDetails = connectionDetails, data = data)
}
```

Reply:
> I think the context of this class is useful for a different reason that serialization or de-serialization. 

That's a fair point, but this PR seems to set the basis for one example of implementing the other R6 classes.  I was looking at the R6 classes as a means to define the analytic input API (at least at a data structure level, not a functional level).  So there will be some that are pure data oriented (to define structure and handle serialization) and maybe there will be some that have behavior (for example, CI has a query builder R6 class I believe).  

Maybe we need to see an example of both in this PR so we see where lines are drawn between behavior classes and structure classes.